src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CheckForPoison.cs

```diff
@@ -136,8 +141,13 @@ public class CheckForPoison : Task
 
         private const string PoisonMarker = "POISONED";
 
+        private IReadOnlyList<string> nonShippingPackages;
+
         public override bool Execute()
         {
+            // Populate the list of non-shipping packages
+            nonShippingPackages = GetAllNonShippingPackages();
```

Comment:
Why is the instance field necessary?  Couldn't this get passed into GetPoisonedFiles or invoked as part of GetPoisonedFiles?

Reply:
Right - this was left from an early code - will refactor.